### PR TITLE
Add schemas for Microsoft.DataFactory resources

### PR DESCRIPTION
**Pull request by bgold09**
* Adds schemas for data factories (the top-level resource in the
    Microsoft.DataFactory namespace) and its child resources (linked
    services, datasets and pipelines).
  * We are not supporting template expressions at this time.
    This commit is aimed at supporting the "export resource" feature.

### CODE REVIEW

**Comment by bgold09** [comment]
@TianoMS this is the new version of our schemas

**Comment by TianoMS** [comment]
>         },

We don't advise RP to put location here, as it's very difficult to keep this list consistent with your service. Our suggestion is not define this list and just fall back to ARM default location list. 

---
Refers to: schemas/2015-10-01/Microsoft.DataFactory.json:58 in 7948879. [](commit_id = 794887934999fd21a931e1c30be0ec39802846cb, deletion_comment = False)

**Comment by TianoMS** [comment]
>         }

This is not a supported property in resource. 

---
Refers to: schemas/2015-10-01/Microsoft.DataFactory.json:74 in 7948879. [](commit_id = 794887934999fd21a931e1c30be0ec39802846cb, deletion_comment = False)

**Comment by TianoMS** [comment]
>         }

same here. 

---
Refers to: schemas/2015-10-01/Microsoft.DataFactory.json:152 in 7948879. [](commit_id = 794887934999fd21a931e1c30be0ec39802846cb, deletion_comment = False)

**Comment by bgold09** [comment]
@TianoMS comments addressed

File: schemas/2015-01-01/deploymentTemplate.json [hunk 1]
```diff
@@ -148,6 +148,10 @@
                   { "$ref": "http://schema.management.azure.com/schemas/2015-10-31/Microsoft.Automation.json#/resourceDefinitions/jobs" },
                   { "$ref": "http://schema.management.azure.com/schemas/2015-10-31/Microsoft.Automation.json#/resourceDefinitions/jobSchedules" },
                   { "$ref": "http://schema.management.azure.com/schemas/2015-10-01/Microsoft.Media.json#/resourceDefinitions/mediaServices" },
+                  { "$ref": "http://schema.management.azure.com/schemas/2015-10-01/Microsoft.DataFactory.json#/resourceDefinitions/dataFactories" },
+                  { "$ref": "http://schema.management.azure.com/schemas/2015-10-01/Microsoft.DataFactory.json#/resourceDefinitions/linkedServices" },
+                  { "$ref": "http://schema.management.azure.com/schemas/2015-10-01/Microsoft.DataFactory.json#/resourceDefinitions/datasets" },
+                  { "$ref": "http://schema.management.azure.com/schemas/2015-10-01/Microsoft.DataFactory.json#/resourceDefinitions/pipelines" },
```
Comment (TianoMS) [inline schemas/2015-01-01/deploymentTemplate.json]: Can we add these to schemas/2014-04-01-preview/deploymentTemplate.json too?

**Comment by TianoMS** [comment]
@bgold09 Looks good. Please make changes to schemas/2014-04-01-preview/deploymentTemplate.json.

File: schemas/2015-01-01/deploymentTemplate.json (hunk 1 above)
Comment (bgold09) [inline schemas/2015-01-01/deploymentTemplate.json]: @TianoMS added to the other deployment template file

**Pull request merged**